peer.go

```diff
@@ -978,14 +980,68 @@ func (p *peer) handleRemoteClose(req *lnwire.CloseRequest) {
 		return
 	}
 
-	// TODO(roasbeef): also wait for confs before removing state
-	peerLog.Infof("ChannelPoint(%v) is now "+
-		"closed", key)
-	if err := wipeChannel(p, channel); err != nil {
-		peerLog.Errorf("unable to wipe channel: %v", err)
+	// Start a go routine that will mark the channel as fully closed after one
+	// confirmation
+	updateChan := make(chan *lnrpc.CloseStatusUpdate)
+	errChan := make(chan error)
+	txid := closeTx.TxHash()
+
+	go func() {
+		notifier := p.server.chainNotifier
+		confNtfn, err := notifier.RegisterConfirmationsNtfn(&txid, 1)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		select {
+		case _, ok := <-confNtfn.Confirmed:
+			if !ok {
+				return
+			}
+			peerLog.Infof("ChannelPoint(%v) is now "+
+				"fully closed", key)
+			if err := channel.MarkChannelAsFullyClosed(); err != nil {
+				peerLog.Errorf("unable to mark channel fully closed: %v", err)
+			}
+
+			p.server.breachArbiter.settledContracts <- &req.ChannelPoint
+		case <-p.quit:
+			return
+		}
+
+		updateChan <- &lnrpc.CloseStatusUpdate{
```

Comment:
In this context, we're handling a channel closure initiated by  the _remote_ party. Therefore, there's no RPC client to send notifications to.